cli/daemon/export/export.go

```diff
@@ -0,0 +1,272 @@
+		if err != nil {
+			log.Error().Err(err).Msg("invalid image tag")
+			return false, nil
+		}
+		log.Info().Msg("pushing image to docker registry")
+		if err := pushDockerImage(ctx, log, img, tag); err != nil {
+			log.Error().Err(err).Msg("unable to push docker image")
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+// buildDockerImage builds a docker image.
+func buildDockerImage(ctx context.Context, log zerolog.Logger, req *daemonpb.ExportRequest, res *compiler.Result) (v1.Image, error) {
+	log.Info().Msg("building docker image")
+	opener, err := buildImageFilesystem(ctx, res)
+	if err != nil {
+		return nil, errors.Wrap(err, "build image fs")
+	}
+
+	layer, err := tarball.LayerFromOpener(opener,
+		tarball.WithEstargz,
+		tarball.WithEstargzOptions(
+			estargz.WithPrioritizedFiles([]string{appExePath}),
+		),
+		tarball.WithCompressedCaching,
+		tarball.WithCompressionLevel(5), // balance speed and compression
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "create tarball layer")
+	}
+
+	created := v1.Time{Time: time.Now()}
+	img, err := mutate.Append(empty.Image, mutate.Addendum{
+		Layer: layer,
+		History: v1.History{
+			Author:    "encore-app",
+			Created:   created,
+			CreatedBy: "encore.dev",
+			Comment:   "Built with encore.dev, the backend development engine",
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "add layer")
+	}
+
+	cfg, err := img.ConfigFile()
+	if err != nil {
+		return nil, errors.Wrap(err, "get image config")
+	}
+	cfg = cfg.DeepCopy()
+	cfg.Config.Entrypoint = []string{appExePath}
+	cfg.Config.Cmd = nil
+	cfg.Author = "encore.dev"
+	cfg.Created = created
+	cfg.Architecture = req.Goarch
+	cfg.OS = req.Goos
+
+	img, err = mutate.ConfigFile(img, cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "add config")
+	}
+
+	log.Info().Msg("successfully built docker image")
+	return img, nil
+}
+
+func buildImageFilesystem(ctx context.Context, res *compiler.Result) (opener tarball.Opener, err error) {
+	tarFile, err := os.CreateTemp("", "docker-img")
+	if err != nil {
+		return nil, errors.Wrap(err, "mktemp")
+	}
+	defer func() {
+		if e := tarFile.Close(); e != nil && err == nil {
+			err = errors.Wrap(e, "close docker-img file")
+		}
+	}()
+
+	tw := tar.NewWriter(tarFile)
+
+	{
+		appFile, err := os.Open(res.Exe)
+		if err != nil {
+			return nil, errors.Wrap(err, "open")
+		}
+		defer func() { _ = appFile.Close() }()
+
+		fi, err := appFile.Stat()
+		if err != nil {
+			return nil, errors.Wrap(err, "stat")
+		}
+		err = tw.WriteHeader(&tar.Header{
+			Name:     appExePath,
+			Typeflag: tar.TypeReg,
+			Size:     fi.Size(),
+			Mode:     0555,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "add file to tar")
+		}
+		if _, err := io.Copy(tw, appFile); err != nil {
+			return nil, errors.Wrap(err, "copy file to tar")
+		}
+	}
+
+	// Download ca certs
+	const certsDest = "/etc/ssl/certs/ca-certificates.crt" // from https://go.dev/src/crypto/x509/root_linux.go
```

Comment:
This comment says it's from `go.dev` - but then downloads them Mozilla - I think this comment is a little redundant - as that's the standard ca cert location

Reply:
The reason we chose that destination folder is sourced from that link; did you confuse it with where we download the certs from?

Reply:
At first yes, then I realised what you meant in this comment - but I thought it was redundant and potentially confusing - as `/etc/ssl/certs/ca-certificates.crt` isn't Go specific but more general for linux